src/ui/components/Library/Navigation/TeamButton.tsx

```diff
@@ -43,25 +45,73 @@ export function TeamButton({
   };
 
   return (
-    <Link
-      href={url}
+    <div
       className={classNames(
-        `${styles.teamRow} group flex flex-row justify-between space-x-2 px-4 py-2 text-left transition duration-200 hover:text-white focus:outline-none`,
-        isSelected ? `${styles.teamRowActive} cursor-auto` : "cursor-pointer"
+        `flex flex-col ${styles.teamRow} `,
+        isSelected ? `${styles.teamRowActive}` : ""
       )}
-      onClick={onClick}
     >
-      <span className="overflow-hidden overflow-ellipsis whitespace-pre">
-        <div className="flex">
-          <LibraryIcon teamType={isTest ? "tests" : "team"} />
-          {label}
+      <Link
+        href={url}
+        className={classNames(
+          `group flex flex-row justify-between space-x-2 px-4 py-2 text-left transition duration-200 hover:text-white focus:outline-none`,
+          isSelected ? `cursor-auto` : "cursor-pointer"
+        )}
+        onClick={onClick}
+      >
+        <span className="overflow-hidden overflow-ellipsis whitespace-pre">
+          <div className="flex">
+            <LibraryIcon teamType={isTest ? "tests" : "team"} />
+            {label}
+          </div>
+        </span>
+        {isNew ? (
+          <div className={"rounded-md bg-primaryAccent px-3 py-0.5 text-xs text-white"}>New</div>
+        ) : null}
+        {showSettingsButton ? <SettingsButton /> : null}
+      </Link>
+      {isSelected ? <TestTeamViews /> : null}
+    </div>
+  );
+}
+
+// We don't currently have a context shared between the navigation
+// and the content of the library UI. Getting the view/teamId this
+// way is pretty clunky, so we should probably lift that state up
+function TestTeamViews() {
+  const [showTestsView] = useLocalStorageUserData("enableTestView");
+  const router = useRouter();
+  const cont = useContext(ViewContext);
+  const view = useGetTeamRouteParams().view;
+  const { teamId } = useGetTeamRouteParams();
+
+  const setView = (view: View) => {
+    pushRoute(router, `/team/${teamId}/${view}`);
+  };
+
+  return (
+    <div className="pl-4 pb-2">
+      <div className="flex flex-col">
+        <div
+          className={`px-4 py-1 ${
+            view === "runs" ? "font-bold" : ""
+          } hover:cursor-pointer hover:font-bold`}
```

Comment:
I think we should probably also avoid using "bold" style for hover state. Having text resize on mouse over feels jarring.

```suggestion
          } hover:cursor-pointer`}
```